### PR TITLE
When calling chown, use : instead of . to separate the user and group for cross platform compatibility.

### PR DESCRIPTION
**Pull request by roberthbailey**
**What this PR does / why we need it**: Makes it possible to build on a Mac, which was broken by #39515. 

**Special notes for your reviewer**:

**Release note**:
`NONE`

### CODE REVIEW

File: build/common.sh [hunk 1]
```diff
@@ -411,7 +411,7 @@ function kube::build::clean() {
 function kube::build::build_image() {
   mkdir -p "${LOCAL_OUTPUT_BUILD_CONTEXT}"
   # Make sure the context directory owned by the right user for syncing sources to container.
-  chown -R ${USER_ID}.${GROUP_ID} "${LOCAL_OUTPUT_BUILD_CONTEXT}"
+  chown -R ${USER_ID}:${GROUP_ID} "${LOCAL_OUTPUT_BUILD_CONTEXT}"
```
Comment (ixdy) [inline build/common.sh]: L502 or so needs a fix too, I think

Comment (roberthbailey) [inline build/common.sh]: oops, fixed that one and re-pushed.

**Comment by roberthbailey** [comment]
I thought I did follow the release note process....

Also, since I'm an owner of the cluster directory, should it really require another owner to approve?

@grodrigues3

**Comment by ixdy** [comment]
@roberthbailey it looks like you change the release note block from ` ```release-note` to just `, which confused it?

**Comment by grodrigues3** [comment]
@roberthbailey nope, it shouldn't.  that feature should be rolled out today.

https://github.com/kubernetes/test-infra/pull/1730/files

**Comment by roberthbailey** [comment]
In that case I'm going to manually label this as approved, so that I can resume building k8s locally.

**Comment by grodrigues3** [comment]
To be clear, as an Approver (in the owners files), you could have also written /approve and that would have achieved the same thing.

**Pull request merged**